client/my-sites/stats/stats-module/index.jsx

```diff
@@ -64,6 +64,25 @@ class StatsModule extends Component {
 		}
 	}
 
+	shouldComponentUpdate( nextProps ) {
+		for ( const key in nextProps ) {
+			if ( this.props[ key ] !== nextProps[ key ] ) {
+				if ( key !== 'data' ) {
+					return true; // found shallow unequal prop, should update.
+				}
+
+				// Special case for data prop, do deep equality check.
+				const isDeepEqual = isEqual( this.props[ key ], nextProps[ key ] );
```

Comment:
Nice - curious was that on a site with a large amount of stats data... specifically referrers?